app/services/migration/migrators/application.rb

```diff
@@ -30,41 +36,86 @@ def model
         :application
       end
 
+      def dependencies
+        %i[cohort lead_provider schedule course user school]
+      end
+
       def ecf_npq_applications
-        Migration::Ecf::NpqApplication.joins(:participant_identity)
+        Migration::Ecf::NpqApplication
+          .joins(:participant_identity)
+          .includes(:cohort, :user, profile: :schedule)
       end
     end
 
     def call
       run_once { report_applications_not_in_ecf_as_failures }
 
       migrate(self.class.ecf_npq_applications) do |ecf_npq_application|
-        application = applications_by_ecf_id[ecf_npq_application.id]
-        raise ActiveRecord::RecordNotFound, "Application not found" unless application
+        application = ::Application.find_by!(ecf_id: ecf_npq_application.id)
+
+        ensure_relationships_are_consistent!(ecf_npq_application, application)
+
+        ecf_schedule = ecf_npq_application.profile&.schedule
+        if ecf_schedule
+          schedule_cohort_id = find_cohort_id!(start_year: ecf_schedule.cohort.start_year)
```

Comment:
hmmm I think it would have been easier for us to have `ecf_id` on schedule to avoid all this, will add a ticket